web/client/codechecker_client/client.py

```diff
@@ -253,51 +223,18 @@ def setup_client(product_url, product_client=False):
                   "looks like this: 'http://my.server.com:80/ProductName'.")
         sys.exit(2)  # 2 for argument error.
 
-    _, session_token = setup_auth_client(protocol, host, port)
-
-    # Check if the product exists.
-    client = setup_product_client(protocol, host, port, product_name=None,
-                                  session_token=session_token)
-    product = client.getProducts(product_name, None)
-    product_error_str = None
-    if not product:
-        product_error_str = "It does not exist."
-    elif len(product) != 1:
-        product_error_str = "Multiple products can be found with the given " \
-                            "name."
-    else:
-        if product[0].endpoint != product_name:
-            # Only a "substring" match was found. We explicitly reject it
-            # on the command-line!
-            product_error_str = "It does not exist."
-
-        elif not product[0].accessible:
-            product_error_str = "You do not have access."
-
-        elif product[0].databaseStatus != shared.ttypes.DBStatus.OK:
-            product_error_str = "The database has issues, or the connection " \
-                                "is badly configured."
-
-    if product_error_str:
-        LOG.error("The given product '%s' can not be used! %s", product_name,
-                  product_error_str)
-        sys.exit(1)
-
-    if product_client:
-        LOG.debug("returning product client")
-        return client
-    else:
-        LOG.debug("returning service client")
-        # Service client was requested, setup
-        # and return it.
-        return setup_service_client(protocol,
-                                    host,
-                                    port,
-                                    product_name,
-                                    session_token)
+    # Check if local token is available.
+    cred_manager = UserCredentials()
+    session_token = cred_manager.get_token(host, port)
 
+    # Local token is missing ask remote server.
+    if not session_token:
+        auth_client = setup_auth_client(protocol, host, port, session_token)
```

Comment:
In this code block the `session_token` variable is `None`, so you don't need to forward it to the `setup_auth_client` function.